packages/xrpl/test/integration/utils.ts

```diff
@@ -1,45 +1,132 @@
 import { assert } from 'chai'
-import _ from 'lodash'
+import omit from 'lodash/omit'
 import { decode } from 'ripple-binary-codec'
-import { Client, Wallet, AccountInfoRequest } from 'xrpl-local'
+import {
+  Client,
+  Wallet,
+  AccountInfoRequest,
+  type SubmitResponse,
+  TimeoutError,
+  NotConnectedError,
+} from 'xrpl-local'
 import { Payment, Transaction } from 'xrpl-local/models/transactions'
 import { hashSignedTx } from 'xrpl-local/utils/hashes'
 
 const masterAccount = 'rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh'
 const masterSecret = 'snoPBrXtMeMyMHUVTgbuqAfg1SUTb'
 
-export async function ledgerAccept(client: Client): Promise<void> {
+// let ledgerAcceptMutex = false
+
+export async function ledgerAccept(
+  client: Client,
+  retries?: number,
+): Promise<void> {
   const request = { command: 'ledger_accept' }
-  await client.connection.request(request)
+
+  try {
+    await client.connection.request(request)
+  } catch (error) {
+    if (retries === undefined) {
+      setTimeout(() => {
+        ledgerAccept(client, 10)
+      }, 1000)
+    } else if (retries > 0) {
+      setTimeout(() => {
+        ledgerAccept(client, retries - 1)
+      }, 1000)
+    } else {
+      throw error
+    }
+  }
 }
 
-export function subscribeDone(client: Client, done: Mocha.Done): void {
+export function subscribeDone(client: Client): void {
   client.removeAllListeners()
-  done()
+}
+
+async function runCommand({
+  client,
+  transaction,
+  wallet,
+  retry = { count: 5, delayMs: 1000 },
+}: {
+  client: Client
+  transaction: Transaction
+  wallet: Wallet
+  retry?: {
+    count: number
+    delayMs: number
+  }
+}): Promise<SubmitResponse> {
+  let response: SubmitResponse
+  try {
+    response = await client.submit(transaction, { wallet })
+
+    while (
+      ['tefPAST_SEQ', 'tefMAX_LEDGER'].includes(
```

Comment:
Adds retry if other commands finish before this current command, messing up sequence numbers.

Reply:
Is there a note to fix the sequence numbers in a future PR?

Reply:
Yes in the followup list